torch/_inductor/fx_passes/pad_mm.py

```diff
@@ -504,6 +547,107 @@ def write_pad():
         return should_pad
 
 
+def get_context(
+    mat1,
+    mat2,
+    mat1_pre_padded,
+    mat2_pre_padded,
+    m_padded_length,
+    k_padded_length,
+    n_padded_length,
+):
+    context = AHContext()
+
+    context.add_feature("m", mat1.shape[0])
+    context.add_feature("k", mat1.shape[1])
+    context.add_feature("n", mat2.shape[1])
+
+    mat1_strides = mat1.stride()
+    mat2_strides = mat2.stride()
+    context.add_feature("mat1_stride_0", mat1_strides[0])
+    context.add_feature("mat1_stride_1", mat1_strides[1])
+    context.add_feature("mat2_stride_0", mat2_strides[0])
+    context.add_feature("mat2_stride_1", mat2_strides[1])
+
+    context.add_feature("m_padded_length", m_padded_length)
+    context.add_feature("k_padded_length", k_padded_length)
+    context.add_feature("n_padded_length", n_padded_length)
+
+    context.add_feature("mat1_iscontig", mat1.is_contiguous(), is_categorical=True)
+    context.add_feature("mat2_iscontig", mat2.is_contiguous(), is_categorical=True)
```

Comment:
nit: `is_contiguous` could be augmented in autoheuristic_utils since you can compute it from strides.

Reply:
I implemented this change and updated the dataset in https://github.com/AlnisM/autoheuristic-datasets accordingly. The generated code for the heuristic is slightly different, but essentially the same, because the order of the categorical features seem to make a difference when they are one-hot encoded.